### PR TITLE
Update process statuses, update process status rendering in tables, d…

### PR DESCRIPTION
**Pull request by burzomir**
…isable process status table filters

Feature #13776

Arvados-DCO-1.1-Signed-off-by: Michal Klobukowski <michal.klobukowski@contractors.roche.com>

### CODE REVIEW

File: src/components/data-table/data-column.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ export interface DataColumn<T, F extends DataTableFilterItem = DataTableFilterIt
     name: string;
     selected: boolean;
     configurable: boolean;
-    sortDirection: SortDirection;
+    sortDirection?: SortDirection;
```
Comment (unodgs) [inline src/components/data-table/data-column.ts]: why `?` ?

File: src/store/context-menu/context-menu-actions.ts [hunk 2]
```diff
@@ -87,13 +88,10 @@ export const openProcessContextMenu = (event: React.MouseEvent<HTMLElement>) =>
     (dispatch: Dispatch, getState: () => RootState) => {
         const { location } = getState().router;
```
Comment (unodgs) [inline src/store/context-menu/context-menu-actions.ts]: why not simply: `const location = getState().router.location`?

File: src/components/data-table/data-column.ts (hunk 1 above)
Comment (burzomir) [reply src/components/data-table/data-column.ts]: Sort direction is meant to control sort direction and whether a column can be sorted

File: src/store/context-menu/context-menu-actions.ts (hunk 2 above)
Comment (burzomir) [reply src/store/context-menu/context-menu-actions.ts]: IMO simpler is to write location once rather than twice.

Comment (unodgs) [reply src/store/context-menu/context-menu-actions.ts]: OK It's a matter of personal preference actually :)

**Pull request merged**